lib/storage/src/chunks/getChunkStream.ts

```diff
@@ -0,0 +1,43 @@
+import { RawDataPart } from "../Upload";
+import { Buffer } from "buffer";
+
+interface Buffers {
+  chunks: Buffer[];
+  length: number;
+}
+
+export async function* getChunk<T>(
+  data: T,
+  partSize: number,
+  getNextData: (data: T) => AsyncGenerator<Buffer>
+): AsyncGenerator<RawDataPart> {
+  let partNumber = 1;
+  const currentBuffer: Buffers = { chunks: [], length: 0 };
+
+  for await (const datum of getNextData(data)) {
+    currentBuffer.chunks.push(datum);
+    currentBuffer.length += datum.length;
+
+    while (currentBuffer.length >= partSize) {
+      // Concat all the buffers together once.
+      let dataChunk: Buffer = currentBuffer.chunks[0];
+      if(currentBuffer.chunks.length) {
```

Comment:
looks like prettier is not run on this file. Commenting to verify prettier is run on all files.
Previous discussion on this issue: https://github.com/aws/aws-sdk-js-v3/issues/1888

```suggestion
      if (currentBuffer.chunks.length) {
```